### DISCUSSION TITLE
SFTPGO with openID Connect integration

### DISCUSSION THREAD

**Discussion by devrime** [drakkan/sftpgo#785] [Q&A] [answered]
I have used  drakkan/sftpgo:2.2.x container image. I have configured keycloak as described. I have started sftpgo container with below parameters;
environment variables;
      - SFTPGO_DATA_PROVIDER__DRIVER=postgresql
      - SFTPGO_DATA_PROVIDER__NAME=sftpgo
      - SFTPGO_DATA_PROVIDER__HOST=postgresql-sftpgo
      - SFTPGO_DATA_PROVIDER__PORT=5432
      - SFTPGO_DATA_PROVIDER__USERNAME=sftpgo
      - SFTPGO_DATA_PROVIDER__PASSWORD=sftpgo
      - SFTPGO_DATA_PROVIDER__SSLMODE=0
      - SFTPGO_HTTPD__BINDINGS__OIDC__CLIENT_ID="sftpgo-client"
      - SFTPGO_HTTPD__BINDINGS__OIDC__CLIENT_SECRET="aAl4DVwhquXCX2LDKL0J9O0dVMXEo1Ga"
      - SFTPGO_HTTPD__BINDINGS__OIDC__CONFIG_URL=http://keycloak:10991/auth/realms/sftpgo
      - SFTPGO_HTTPD__BINDINGS__OIDC__REDIRECT_BASE_URL=http://sftpgo:10200
      - SFTPGO_HTTPD__BINDINGS__OIDC__USERNAME_FIELD="preferred_username",
      - SFTPGO_HTTPD__BINDINGS__OIDC__ROLE_FIELD="sftpgo_role"      
    volumes:
      - sftpgo-data:/srv/sftpgo
      - sftpgo-home:/var/lib/sftpgo

I have followed below article when configuring keycloak and sftpgo
https://github.com/drakkan/sftpgo/blob/main/docs/oidc.md

I could not see the "login with openid connect" button on sftpgo login page.
Could you please help me?
      -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -       -

**Comment by drakkan** [comment, marked as answer]
You need to use the development version, please try an edge based docker variant

**Reply by devrime** [reply]
I have tried drakkan/sftpgo:edge version and it is the same. I could not see the "login with openid connect" button

**Reply by drakkan** [reply]
`SFTPGO_HTTPD__BINDINGS__OIDC__CLIENT_ID` must be `SFTPGO_HTTPD__BINDINGS__0__OIDC__CLIENT_ID` and so on, bindings are a list. You can see the loaded configuration options in the SFTPGo logs

**Reply by devrime** [reply]
ok for now  environment variables are taken from sftpgo
thanks
